packages/wrangler/src/pages/functions/buildWorker.ts

```diff
@@ -164,3 +148,135 @@ export function buildWorker({
 		}
 	);
 }
+
+export type RawOptions = {
+	workerScriptPath: string;
+	outfile: string;
+	directory: string;
+	minify?: boolean;
+	sourcemap?: boolean;
+	watch?: boolean;
+	plugins?: Plugin[];
+	onEnd?: () => void;
+	buildOutputDirectory?: string;
+	nodeCompat?: boolean;
+	local: boolean;
+	betaD1Shims?: string[];
+};
+
+/**
+ * This function bundles a raw `_worker.js` Pages file
+ * before it gets deployed.
+ *
+ * This allows Wrangler to add shims and other wrappers
+ * around the handlers, which is useful to support beta features.
+ */
+export function buildRawWorker({
+	workerScriptPath,
+	outfile,
+	directory,
+	minify = false,
+	sourcemap = false,
+	watch = false,
+	plugins = [],
+	onEnd = () => {},
+	nodeCompat,
+	local,
+	betaD1Shims,
+}: RawOptions) {
+	return bundleWorker(
+		{
+			file: workerScriptPath,
+			directory,
+			format: "modules",
+		},
+		resolve(outfile),
+		{
+			minify,
+			sourcemap,
+			watch,
+			nodeCompat,
+			loader: {
+				".txt": "text",
+				".html": "text",
+			},
+			define: {},
+			betaD1Shims: (betaD1Shims || []).map(
+				(binding) => `${D1_BETA_PREFIX}${binding}`
+			),
+			plugins: [...plugins, buildNotifierPlugin(onEnd)],
+			isOutfile: true,
+			serveAssetsFromWorker: false,
+			disableModuleCollection: true,
+			rules: [],
+			checkFetch: local,
+			targetConsumer: local ? "dev" : "publish",
+			local,
+			experimentalLocal: false,
+		}
+	);
+}
+
+/**
+ * Creates an esbuild plugin that can notify Wrangler (via the `onEnd()`)
+ * when the build completes.
+ */
+export function buildNotifierPlugin(onEnd: () => void): Plugin {
+	return {
+		name: "wrangler notifier and monitor",
+		setup(pluginBuild) {
+			pluginBuild.onEnd((result) => {
+				if (result.errors.length > 0) {
+					logger.error(
+						`${result.errors.length} error(s) and ${result.warnings.length} warning(s) when compiling Worker.`
+					);
+				} else if (result.warnings.length > 0) {
+					logger.warn(
+						`${result.warnings.length} warning(s) when compiling Worker.`
+					);
+					onEnd();
+				} else {
+					logger.log("✨ Compiled Worker successfully");
+					onEnd();
+				}
+			});
+		},
+	};
+}
+
+/**
+ * Runs the script through a simple esbuild bundle step to check for unwanted imports.
+ *
+ * This is useful when the user chooses not to bundle the `_worker.js` file by setting
+ * `--no-bundle` at the command line.
+ */
+export async function checkRawWorker(scriptPath: string, onEnd: () => void) {
```

Comment:
same comment here...if we're running this fn only on `_worker.js`, I'd rename `RawWorker` to `WorkerJS`.